Intersect.Client.Core/Networking/PacketHandler.cs

```diff
@@ -866,6 +866,11 @@ public void HandlePacket(IPacketSender packetSender, MapEntityVitalsPacket packe
                     Globals.Me.CombatTimer = Timing.Global.Milliseconds + en.CombatTimeRemaining;
                 }
             }
+
+            if (entity.Type == EntityType.Resource)
+            {
+                ((Resource)entity).UpdateCurrentState();
```

Comment:
this should be `entity is Resource resourceEntity` and then use `resourceEntity` instead of casting